data/archiver/archiver.py

```diff
@@ -0,0 +1,102 @@
+import os
+import logging
+from data.archiver.aws_s3_client import AwsS3
+from data.archiver.ftp_uploader import FtpUploader
+from data.archiver.utils import compress, md5
+from data.archiver.dataclass import DataArchiverRequest, DataArchiverResult
+from data.archiver.ingest_api import Ingest
+
+STREAM_ARCHIVE = False
+
+class Archiver:
+
+    def __init__(self):
+        self.ingest_api = Ingest()
+        self.aws_client = AwsS3()
+        self.ftp = None
+        
+        self.logger = logging.getLogger(__name__)
+        self.logger.setLevel(logging.INFO)
+
+    def start(self, req: DataArchiverRequest):
+
+        self.logger.info(f'Getting sequence files for submission {req.sub_uuid} from INGEST_API')
+        submission_files = self.ingest_api.get_sequence_files(req.sub_uuid)
+        self.logger.info(submission_files)
+
+        if not submission_files:
+            return DataArchiverResult.error_res('No sequence files for submission')
+
+        if not req.files:
+            # archive all files
+            req.files = submission_files
+        
+        self.logger.info(req)
+        res = DataArchiverResult.empty(req)
+
+        try:
+            # TODO logic here to decide between local copy or stream archive/upload to ena
+            if STREAM_ARCHIVE:
+                return
+            else:
+                result = self.archive_files_via_localcopy(req, res)
+                self.logger.info(result)
+                return result
+
+        except Exception as ex:
+            print(ex)
+
+    def archive_files_via_localcopy(self, req: DataArchiverRequest, res: DataArchiverResult):
+
+        self.logger.info(f'# step 1 download sequence files to be uploaded to FTP from S3')
+        self.aws_client.get_files(req, res)
+
+        self.logger.info(f'# step 2 compress files using gz')
+        def try_compress(f):
```

Comment:
Why is this a function within a function?

Reply:
What is `f`? This is an unclear variable name.

Reply:
Python inner function, not used elsewhere, closer to usage. Changed `f` -> `file`